drivers/mge-hid.c

```diff
@@ -267,27 +275,55 @@ static const char *eaton_abm_status_fun(double value)
 
 	upsdebugx(2, "ABM numeric status: %i", (int)value);
 
-	switch ((long)value)
+	if (dstate_getinfo("battery.charger.type"))
```

Comment:
I don't really understand this `if`-section. It seems here the decision is made for two different kinds of ABM handling, but based on the existence of a variable `battery.charger.type`? Why not `advanced_battery_monitoring == ABM_ENABLED_TYPE` that we already know? Does the existence of `battery.charge.type` imply a different kind of ABM is used and can we ensure that we're not sending other devices down the wrong road if basing this on some variable existence alone? Because this seems to say if that variable exists `2 = floating` and not `2 = discharging`.